### DISCUSSION TITLE
(Web3Modal) Access to Trust Wallet in BNB chain

### DISCUSSION THREAD

**Discussion by JoaoArturCosta** [WalletConnect/walletconnect-monorepo#2323] [Developer Support]
I'm trying to set up wallet connect to interact with the BNB chain on iphone, but when opening the modal I get a bunch of wallets available but not TrustWallet or MetaMask... Is this intended is there no support for these wallets on the Iphone ?

Btw, on Android and the browser it works, only on Iphone it is missing.

**Comment by crypblizz8** [comment]
Hey there, could you share some screenshots of what you are experiencing?

**Reply by JoaoArturCosta** [reply]
Hi, this is what I mean 
![image](https://user-images.githubusercontent.com/23420227/237061946-afecef28-6131-4177-9fba-4780a924232b.png)

In IOS, I don't have the trust wallet option available on any browser

**Comment by finessevanes** [comment]
That screenshot looks like it's an Android, I don't see MetaMask or Trust. Can you also share your code implementation?

**Reply by JoaoArturCosta** [reply]
I can assure you it's ios, on android you don't select which wallet you want it just shows the installed ones

here is the code const connect = async ({ chainId, web3Connection }) => {
  try {
    const provider = await createProvider();

    await provider.connect({
      namespaces: {
        eip155: {
          chains: [
            `eip155:${chainId}`,
          ],
          events: [
            'accountsChanged',
            'chainChanged',
            'connect',
            'disconnect',
            'message',
          ],
          methods: [
            'eth_accounts',
            'eth_sendTransaction',
            'eth_sign',
            'eth_signTransaction',
            'eth_signTypedData',
            'eth_signTypedData_v4',
          ],
          // What I had to dig through WC's code to find this...
          rpcMap: {
            [chainId]: web3Connection,
          },
        },
      },
      skipPairing: true,
    });

    const session = await provider.client.connect({
      requiredNamespaces: provider.namespaces,
    });

    if (session.uri) {
      // Adapted from UniversalProvider.pair() method
      // https://github.com/WalletConnect/walletconnect-monorepo/blob/v2.0/providers/universal-provider/src/UniversalProvider.ts#L126
      provider.uri = session.uri;
      provider.events.emit('display_uri', session.uri);

      const { Web3Modal } = (await import('@web3modal/standalone'));
      const web3Modal = new Web3Modal({ projectId });
      await web3Modal.openModal({
        standaloneChains: [ `eip155:${chainId}` ],
        uri: session.uri,
      });
      provider.session = await session.approval();
      web3Modal.closeModal();

      // Adapted from UniversalProvider.pair() method
      provider.onSessionUpdate();

      const accounts = provider.session?.namespaces?.eip155?.accounts;

      if (accounts?.length) {
        wallet = await providerWeb3(provider);
        return wallet?.web3Provider;
      }
    }
  }
  catch (ex) {
    // eslint-disable-next-line
    console.error(ex);
  }

  return null;
};
